nats.go

```diff
@@ -1634,6 +1649,13 @@ func (nc *Conn) createConn() (err error) {
 		return ErrNoServers
 	}
 
+	// If a custom conn has been provided then use that.
```

Comment:
Curious why not before selecting current server above? Is it because rest of code would fail if currentServer() was never invoked? If so, would not that be a problem if user does not specify any URL? (although I think that we add the localhost:4222 internally by default if none is specified).

Reply:
In a bit of local testing it doesn't appear to matter, so I'll move it up to before `currentServer()`.

Reply:
And i think that `doReconnect()` would not work since we try to get a server from the pool and if we don't have any more we exit the reconnect loop.

Reply:
In which case is it better just to handle this case in `connect()` instead, and only loop through servers in the usual way if `nc.Opts.Conn` isn't set?

Reply:
Actually reconnect (as it stands today) won't work at all, because you would need to reassign a new s.InProcessConn(). By definition, if there is a disconnect it means that one side (or both) of the connection is closed, so it needs to be reassigned a new one.

Reply:
So the issue is not the server pool, since it has 1 URL (always), so it would find one, but then we call createConn() that uses the option one but this connection is no longer valid.

Reply:
I suppose this is unavoidable unless we give a reference to the NATS Server directly (via something like an `interface { InProcessConn() (net.Conn, error)` or similar) so that we can attempt to request a new in-process conn, rather than just expecting to be given a `net.Conn`?

Reply:
I guess that would be better. Plus, it would even simply the steps when creating the connection since you would not have to directly call s.InProcessConn() and then pass it to nats.Conn().

Reply:
Have added this so that you can `nc, err := natsclient.Connect("", natsclient.InProcessServer(natsserver))` now.